grakn-graph/src/main/java/ai/grakn/graph/internal/Cache.java

```diff
@@ -114,6 +118,9 @@ void ifPresent(Consumer<V> modifier){
      * that it can be accessed via all transactions.
      */
     void flush(){
-        if(isPresent()) sharedValue = Optional.of(get());
+        if(isPresent() && !sharedValue.equals(get())) {
+            this.sharedValue = Optional.of(get());
+        }
```

Comment:
Are you sure this works? `sharedValue` is `Optional<V>` so I don't think you can compare it to `get()` which is just `V`.